packages/flutter/lib/src/widgets/slotted_render_object_widget.dart

```diff
@@ -231,14 +241,43 @@ class SlottedRenderObjectElement<S> extends RenderObjectElement {
     }(), '${widget.runtimeType}.slots must not change.');
     assert(slottedMultiChildRenderObjectWidgetMixin.slots.toSet().length == slottedMultiChildRenderObjectWidgetMixin.slots.length, 'slots must be unique');
 
+    final Map<Key, S> oldKeyedSlots = _keyToSlot;
+    _keyToSlot = <Key, S>{};
+
+    // New slots for keyed children. It shouldn't be updated in place because
+    // two children could swap slots.
+    final Map<S, Element> slotToKeyedChild = <S, Element>{};
+    final List<MapEntry<S, Widget?>> unkeyedWidgets = <MapEntry<S, Widget?>>[];
```

Comment:
Can this just be a map? Maybe a LinkedHashMap if insert order needs to be preserved.